README.md

```diff
@@ -45,7 +43,7 @@ fn main() {
         age: 91,
     };
 
-    let cbor_encoded = encode_generic(DagCborCodec, &tree);
+    let cbor_encoded = encode_generic::<DagCborCodec, Tree>(DagCborCodec, &tree);
```

Comment:
I'm surprised this was necessary.

Reply:
It's probably not; it's having a hard time picking up the codec in the doctests and I'm troubleshooting. Because this spans multiple repos I need to push to git, so I've flipped the PR to `Draft`

Reply:
> this spans multiple repos

Multiple repos with a dependency cycle, even (on the `dev-dependencies`)

Reply:
Actually, I know that it's not required because in the library that I'm actually working on I now actually need _fewer_ type annotations. For example

```rust
//Before
let bytes = <DagCborCodec as Codec<Ipld>>::encode_to_vec(dag).unwrap();

// After
let bytes = DagCborCodec.encode_to_vec(dag).unwrap();
```